windows-agent/internal/daemon/daemon_test.go

```diff
@@ -158,6 +158,113 @@ func TestStartQuit(t *testing.T) {
 	}
 }
 
+func TestRestart(t *testing.T) {
```

Comment:
I saw restart (even though it's private) as complex as Quit, thus I considered having few test cases checking its interlocking with contexts, with Quit itself, with the server not running yet, stuff that's more complicated to reproduce in the upper levels.

I managed to end up testing the entire logic of restarting the daemon when the WSL adapter shows up afterwards, in TestAddingWSLAdapterRestarts, having a mock that simulates the OS trigger and cause the restart.
When that happens the mock that provides the adapter IP address is not in sync with the mock providing the notification, though, meaning its list of adapters is set early in tests and remain unchanged. I didn't invest more time to augment that mock, so the current tests would loop forever if I didn't stop the daemon. There are many ways to solve that to augment that test case, but it doesn't seem necessary and this PR is already way too big :)

I'm moving to `internal_test.go` as you suggested.

Reply:
Thansk for moving it! My main intent was indeed to avoid making it seemingly part of the API when looking at the tests.

I agree with you that after separated tests for this is nice to spot any issue around it. As said, I understand the logic to test the outer functionality (basically that we detect and call restart at the right time which then resolve the issue) is difficult, hence my suggestion on real end to end tests for this, which will get into that situation (new machine, no WSL network adapter, and so on…)